CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.18)
+
+project( 
+    inchi 
+    VERSION 1.07
+)
+
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
+set( CMAKE_C_STANDARD 11 )
+
+# this works but maybe it's preferable to only export 
+# the symbols that are needed to consume the library
+set( CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON )
+
+add_subdirectory( INCHI-1-SRC/INCHI_BASE/src )
+add_subdirectory( INCHI-1-SRC/INCHI_API/libinchi/src )
+add_subdirectory( INCHI-1-SRC/INCHI_API/demos/inchi_main/src )
+add_subdirectory( INCHI-1-SRC/INCHI_API/demos/mol2inchi/src )
+add_subdirectory( INCHI-1-SRC/INCHI_API/demos/test_ixa/src )
+add_subdirectory( INCHI-1-SRC/INCHI_EXE/inchi-1/src )
+
+add_custom_target( run-tests 
```

Comment:
I will add a target for running the tests against the shared library as well. 
Since this will require some refactoring of the test environment I'd prefer to do so in another PR in order to not bloat the present one.